internal/db/copy.go

```diff
@@ -42,3 +40,71 @@ func CopyAllObjects(ctx context.Context, tx pgx.Tx, source int64, target int64)
 
 	return nil
 }
+
+func CloneToProject(ctx context.Context, tx pgx.Tx, source int64, target int64, vrange VersionRange) (*int64, error) {
+	objectQuery := &pb.ObjectQuery{
+		Path:        "",
+		IsPrefix:    true,
+		WithContent: false,
+		WithHash:    true,
+	}
+
+	latestVersion, err := LockLatestVersion(ctx, tx, target)
+
+	if err != nil {
+		return nil, fmt.Errorf("copy to project could not lock target (%d) version: %w", target, err)
+	}
+
+	newTargetVersion := latestVersion + 1
+
+	builder := newQueryBuilder(source, vrange, objectQuery)
+	removedSql, removedArgs := builder.buildRemoved()
+	removedArgsLength := len(removedArgs)
+
+	sql := fmt.Sprintf(`
+	%v
+	UPDATE dl.objects
+	SET stop_version = $%d
+	FROM removed_objects
+	WHERE "removed_objects".path = "objects".path
+	AND "objects".project = $%d
+	AND "objects".stop_version IS NULL
```

Comment:
Nit: Also the pattern in this codebase is to ident SQL one more time. And look for examples of how `WHERE` & `AND` are formatted elsewhere in the codebase.

```
WHERE x = 1
  AND y = 2
```